shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -109,8 +110,15 @@ public void onFlutterUiNoLongerDisplayed() {
       };
 
   FlutterActivityAndFragmentDelegate(@NonNull Host host) {
+    this(host, null);
+  }
+
+  FlutterActivityAndFragmentDelegate(@NonNull Host host, @Nullable FlutterEngineGroup engineGroup) {
     this.host = host;
     this.isFirstFrameRendered = false;
+    if (engineGroup != null) {
+      this.engineGroup = engineGroup;
+    }
```

Comment:
nit: Removing the if judgment will make the code simpler

```suggestion
    this.engineGroup = engineGroup;
```

Reply:
done